service/history/execution/mutable_state_util_test.go

```diff
@@ -148,3 +148,62 @@ func TestFindAutoResetPoint(t *testing.T) {
 	})
 	assert.Equal(t, pt, pt5)
 }
+
+func TestTrimBinaryChecksums(t *testing.T) {
+	testCases := []struct {
+		name           string
+		maxResetPoints int
+		expected       []string
+	}{
+		{
+			name:           "not reach limit",
+			maxResetPoints: 6,
+			expected:       []string{"checksum1", "checksum2", "checksum3", "checksum4", "checksum5"},
+		},
+		{
+			name:           "reach at limit",
+			maxResetPoints: 5,
+			expected:       []string{"checksum2", "checksum3", "checksum4", "checksum5"},
+		},
+		{
+			name:           "exceeds limit",
+			maxResetPoints: 2,
+			expected:       []string{"checksum5"},
```

Comment:
shouldn't this be checksum4 and checksum5?

Reply:
It needs an extra space for the current run's binChecksum after we trim https://github.com/uber/cadence/pull/5623/files#diff-734db27e2719815a56d959e5cc23ec535d603fdcd499ec5a9b18d0dbf8fa1a55R1988.
It is easier to think the reach at limit case, if we passed in 5 elements and max limit is 5, we will still need to get rid of oldest one.

Reply:
oh I see. just noticed the case above also trims checksum1 and leaves 4 elements. so basically we shrink the list to `maxResetPoints-1`.

Reply:
+1

Reply:
oh, yeah, sorry, was slow to catch this too